src/Bicep.LangServer/Handlers/BicepDefinitionHandler.cs

```diff
@@ -257,6 +257,33 @@ private LocationOrLocationLinks HandlePropertyLocation(DefinitionParams request,
             // Find the underlying VariableSyntax being accessed
             var syntax = result.Origin;
             var propertyAccesses = new List<IdentifierSyntax>();
+            while (true)
+            {
+                if (syntax is PropertyAccessSyntax propertyAccessSyntax)
+                {
+                    propertyAccesses.Insert(0, propertyAccessSyntax.PropertyName);
+                    syntax = propertyAccessSyntax.BaseExpression;
+
+                    continue;
+                }
+
+                if (syntax is TypePropertyAccessSyntax typePropertyAccessSyntax)
+                {
+                    propertyAccesses.Insert(0, typePropertyAccessSyntax.PropertyName);
+                    syntax = typePropertyAccessSyntax.BaseExpression;
+
+                    continue;
+                }
+
+                if (syntax is ParenthesizedExpressionSyntax parenthesized)
+                {
+                    syntax = parenthesized.Expression;
+
+                    continue;
+                }
+
+                break;
+            }
             while (syntax is PropertyAccessSyntax propertyAccessSyntax)
             {
                 // since we are traversing bottom up, add this access to the beginning of the list
```

Comment:
Is this still required?

Reply:
This was meant to be replaced by the `while` block above. Removing in next commit